### ISSUE TITLE
App Refactor - Game like workouts for free

### ISSUE THREAD

**Issue by xKesvaL** [xKesvaL/leetstrength#37]
## tl;dr;
We are ditching firebase and rewriting the app using indexed db ~ we're searching for free solutions we can use for cloud storage.
We are changing the identity of the app to something more game-like with things such as achievements, etc.

## The Database Refactor
This is a complete refactor of the app - We'll ditch firebase.

No authentication will be required, as no *online* profiles will exist anymore, only local data is going to exist. The plan is for us to have no cost and make the app open source and free forever.

### Data Storage
Obviously, the data will only be stored on the used device which is kind of problematic if you lose the device or change devices. As a baseline, we're at least aiming for data exporting / importing and the dream would be to have free cloud synchronization between devices.

## The Experience Refactor
So, it ocurred to me that if we don't have the experience of a social-media like app, we need to have something else catching on. We'll make it a game, altough I don't have an exact plan of how it will look like yet, I really like the idea.

We want something really playful while still good enough for people with high needs. Achievements, rankings... such things.

## To Do

- [ ] Find a wrapper around IndexedDB
- [ ] Ditch firebase everywhere ~ replace with IndexedDB
- [ ] Much more to do...
